indexer/config/config.go

```diff
@@ -85,19 +89,25 @@ type MetricsConfig struct {
 func LoadConfig(path string) (Config, error) {
 	var conf Config
 
-	// Read the config file.
 	data, err := os.ReadFile(path)
 	if err != nil {
 		return conf, err
 	}
 
-	// Replace environment variables.
 	data = []byte(os.ExpandEnv(string(data)))
 
-	// Decode the TOML data.
 	if _, err := toml.Decode(string(data), &conf); err != nil {
 		return conf, err
 	}
 
+	if conf.Chain.Preset != 0 {
```

Comment:
The toml library seems to set everything to 0 if it isn't provided. I looked in the docs for a way to make it explicitly nil but couldn't (quickly) find one. This passes all the tests this implementation though.

Would be nice if we could throw an error when required stuff is not provided fully off of the type. This will throw an error if a unknown config option is passed in though. Maybe worth noting in a github issue that the toml is defaulting a lot of stuff we would prefer to not default

Reply:
Maybe we should make a ticket to provide extra validation to fail early here? We could 
- check all the contract addresses exist on the chain on startup
- - Check that the rpc is up
- - Check that every field is uo

Reply:
what happens if you make the field a pointer to an int? Will it initialize it to `nil` if it's not present?

Reply:
will give it a try